torch/distributed/fsdp/__init__.py

```diff
@@ -17,3 +17,23 @@
     StateDictSettings,
     StateDictType,
 )
+
+__all__ = [
+    "FlatParameter",
```

Comment:
similar comment from @fegin  in https://github.com/pytorch/pytorch/pull/109922, we should not expose `FlatPrameter` as a public API, besides that the rest looks good to me